src/signal_scope/plotwidget.cpp

```diff
@@ -104,6 +107,11 @@ PlotWidget::PlotWidget(PythonChannelSubscriberCollection* subscribers, QWidget *
   QTimer* labelUpdateTimer = new QTimer(this);
   this->connect(labelUpdateTimer, SIGNAL(timeout()), SLOT(updateSignalInfoLabel()));
   labelUpdateTimer->start(100);
+
+  rescalingTimer = new QTimer(this);
+  this->connect(rescalingTimer, SIGNAL(timeout()), SLOT(resetYAxisMaxScale()));
+  yRange[0] = std::numeric_limits<double>::max();
+  yRange[1] = -std::numeric_limits<double>::max();
```

Comment:
small nitpick, can you change these lines to use  "this->yRange"

Also, I think maybe the yRange should be reset to numeric_limits in the function PlotWidget::clearHistory().